src/Bridges/ApplicationLatte/TemplateFactory.php

```diff
@@ -31,14 +31,20 @@ class TemplateFactory implements UI\ITemplateFactory
 	/** @var Nette\Caching\IStorage */
 	private $cacheStorage;
 
+	/** @var string */
+	private $templateClassName;
+
 
 	public function __construct(ILatteFactory $latteFactory, Nette\Http\IRequest $httpRequest = NULL,
-		Nette\Security\User $user = NULL, Nette\Caching\IStorage $cacheStorage = NULL)
+		Nette\Security\User $user = NULL, Nette\Caching\IStorage $cacheStorage = NULL, $templateClassName = Template::class)
 	{
 		$this->latteFactory = $latteFactory;
 		$this->httpRequest = $httpRequest;
 		$this->user = $user;
 		$this->cacheStorage = $cacheStorage;
+
+		\Nette\Utils\Validators::assert($templateClassName, 'string');
```

Comment:
Yeah, it cannot be done like that. It's because `$template` is created during `createTemplate` call (everytime you call it) and therefore you cannot test its instance in constructor. You don't have `ITemplate` in constructor. What implements `ITemplate`? The answer is: created `$template`.